test/test_socketcan_loopback.py

```diff
@@ -0,0 +1,70 @@
+"""
+This module tests sending messages on socketcan with and without local_loopback flag
+
+for a good explanation of why this might be needed:
+https://www.kernel.org/doc/html/v4.17/networking/can.html#socketcan-local-loopback1
+"""
+import unittest
+
+import can
+
+from .config import TEST_INTERFACE_SOCKETCAN
+
+
+@unittest.skipUnless(TEST_INTERFACE_SOCKETCAN, "skip testing of socketcan")
+class LocalLoopbackSocketCan(unittest.TestCase):
+    """ test local_loopback functionality"""
+
+    BITRATE = 500000
+    TIMEOUT = 0.1
+
+    INTERFACE_1 = "socketcan"
+    CHANNEL_1 = "vcan0"
+    INTERFACE_2 = "socketcan"
+    CHANNEL_2 = "vcan0"
+
+    def setUp(self):
+        self._recv_bus = can.Bus(
+            interface=self.INTERFACE_2, channel=self.CHANNEL_2, bitrate=self.BITRATE
+        )
+
+    def tearDown(self):
+        self._recv_bus.shutdown()
+
+    def test_sending_message_with_loopback_enabled(self):
+        """test that sending messages with local_loopback=True produces output even
+        on the local device"""
+        loopback_send_bus = can.Bus(
+            interface=self.INTERFACE_1,
+            channel=self.CHANNEL_1,
+            bitrate=self.BITRATE,
+            local_loopback=True,
+        )
+        try:
+            msg = can.Message(arbitration_id=0x123, is_extended_id=False)
+            loopback_send_bus.send(msg)
```

Comment:
Test that `loopback_send_bus.recv()` returns a message.

Reply:
loopback_send_bus.recv() does NOT return a message. The local_loopback flag (as far as I understand it), means the *other* local sockets open on the same machine will receive the message, but not your own socket. I can instead add an assert that it does not recv() in any case.

Reply:
A bit more info can be read here:

https://www.kernel.org/doc/html/latest/networking/can.html#socketcan-local-loopback1

Reply:
Whoops. OK, that also sounds useful (even more useful). Then yes, do as you said. Maybe briefly document it in `SocketcanBus`?

Reply:
Added. This was very confusing for me also when I started looking at it :)